src/http_event_service.rs

```diff
@@ -0,0 +1,147 @@
+use axum::{debug_handler, extract::State, http::StatusCode, Json};
+use bson::Uuid;
+use log::info;
+use mongodb::Collection;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    foreign_types::{Discount, ProductItem, ProductVariantVersion, ShipmentMethod, TaxRateVersion},
+    user::User,
+};
+
+/// Data to send to Dapr in order to describe a subscription.
+#[derive(Serialize)]
+pub struct Pubsub {
+    #[serde(rename(serialize = "pubsubName"))]
+    pub pubsubname: String,
+    pub topic: String,
+    pub route: String,
+}
+
+/// Reponse data to send to Dapr when receiving an event.
+#[derive(Serialize)]
+pub struct TopicEventResponse {
+    pub status: i32,
+}
+
+/// Default status is `0` -> Ok, according to Dapr specs.
+impl Default for TopicEventResponse {
+    fn default() -> Self {
+        Self { status: 0 }
+    }
+}
+
+/// Relevant part of Dapr event wrapped in a CloudEnvelope.
+#[derive(Deserialize, Debug)]
+pub struct Event {
+    pub topic: String,
+    pub data: EventData,
+}
+
+/// Relevant part of Dapr event.data.
+#[derive(Deserialize, Debug)]
+pub struct EventData {
+    pub id: Uuid,
+}
+
+/// Service state containing database connections.
+#[derive(Clone)]
+pub struct HttpEventServiceState {
+    pub product_variant_version_collection: Collection<ProductVariantVersion>,
+    pub product_item_collection: Collection<ProductItem>,
+    pub tax_rate_version_collection: Collection<TaxRateVersion>,
+    pub discount_collection: Collection<Discount>,
+    pub shipment_method_collection: Collection<ShipmentMethod>,
+    pub user_collection: Collection<User>,
+}
+
+/// HTTP endpoint to list topic subsciptions.
+pub async fn list_topic_subscriptions() -> Result<Json<Vec<Pubsub>>, StatusCode> {
+    let pubsub_product_variant_version = Pubsub {
+        pubsubname: "pubsub".to_string(),
+        topic: "catalog/product-variant/created".to_string(),
+        route: "/on-topic-event".to_string(),
+    };
+    let pubsub_product_item = Pubsub {
+        pubsubname: "pubsub".to_string(),
+        topic: "inventory/product_item/created".to_string(),
```

Comment:
does this use an underscore? (and all the others too)

Reply:
Not sure, i think we should standardize it. Is there already a service that uses snake_case or camelCase?

Reply:
Ah, i see that you have used a dash before for `product-variant`

Reply:
also for tax-rate etc.
so I would be happy if we could keep using dashes

Reply:
Fine for me